src/logic/relation.lean

```diff
@@ -182,10 +182,16 @@ lemma mono {p : α → α → Prop} (hp : ∀ a b, r a b → p a b) : ∀ {a b},
 | a _ refl_gen.refl := by refl
 | a b (single h) := single (hp a b h)
 
+instance : is_refl α (refl_gen r) :=
+⟨@refl α r⟩
+
 end refl_gen
 
 namespace refl_trans_gen
 
+instance : is_refl α (refl_trans_gen r) :=
+⟨@refl α r⟩
```

Comment:
Similarly this likely belongs next to `lemma reflexive_refl_trans_gen : reflexive (refl_trans_gen r)`.

Moving those lemmas up would also be fine by me.